### PR TITLE
Deprecate the DATADOG_ENV environment variable in favor of DD_ENV

### PR DESCRIPTION
**Pull request by wantsui**
## Description

This PR updates the env check in sitecustomize to use **DD_ENV** over **DATADOG_ENV** with the `get_env` helper, which adds a deprecation message if **DATADOG_ENV** is used.

When using **DD_ENV**, follow the [Unified Service Tagging](https://docs.datadoghq.com/getting_started/tagging/unified_service_tagging/) recommendation for the setup to ensure the relevant metrics are associated with traces.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: releasenotes/notes/deprecate-datadog-env-3011aafe83b66e37.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+deprecations:
+  - |
+    Deprecate the DATADOG_ENV environment variable in favor of DD_ENV . 
+    The use of DD_ENV should follow Unified Service Tagging
+    recommendations.
```
Comment (brettlangdon) [inline releasenotes/notes/deprecate-datadog-env-3011aafe83b66e37.yaml:6]: do we have a link we can share to these recommendations?

Not sure "Unified Service Tagging" is going to mean much to most people.

**Review by brettlangdon — DISMISSED** [review]

File: releasenotes/notes/deprecate-datadog-env-3011aafe83b66e37.yaml (hunk 1 above)
Comment (wantsui) [reply releasenotes/notes/deprecate-datadog-env-3011aafe83b66e37.yaml:6]: Does reno support markdown to add links? I couldn't find it in the examples here: https://docs.openstack.org/reno/latest/user/examples.html .

I was thinking of adding a link to https://docs.datadoghq.com/getting_started/tagging/unified_service_tagging/ specifically, but wasn't sure how to format it.

Comment (brettlangdon) [reply releasenotes/notes/deprecate-datadog-env-3011aafe83b66e37.yaml:6]: it is RST not Markdown.

https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#links-to-external-web-pages

You can also validate the rendering in the `build_docs` CircleCI artifacts.

e.g. https://845293-61572326-gh.circle-artifacts.com/0/tmp/docs/release_notes.html

File: releasenotes/notes/deprecate-datadog-env-3011aafe83b66e37.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+deprecations:
+  - |
+    Deprecate the DATADOG_ENV environment variable in favor of DD_ENV .
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/deprecate-datadog-env-3011aafe83b66e37.yaml:4]:
```suggestion
    Deprecate the DATADOG_ENV environment variable in favor of DD_ENV. 
```

File: ddtrace/bootstrap/sitecustomize.py [hunk 3]
```diff
@@ -109,8 +109,9 @@ def update_patched_modules():
 
         patch_all(**EXTRA_PATCHED_MODULES)
 
-    if "DATADOG_ENV" in os.environ:
-        tracer.set_tags({constants.ENV_KEY: os.environ["DATADOG_ENV"]})
+    dd_env = get_env("env")
+    if dd_env:
```
Comment (Kyle-Verhoog) [inline ddtrace/bootstrap/sitecustomize.py:113]: We should probably do a `is not None` instead here but I think all our other env var configuration reading uses the falsy check too so this is consistent.

**Review by Kyle-Verhoog — COMMENTED** [review]
LGTM with just the nit I caught in the release note addressed

**Review by Kyle-Verhoog — APPROVED** [review]

File: releasenotes/notes/deprecate-datadog-env-3011aafe83b66e37.yaml (hunk 1 above)
Comment (wantsui) [reply releasenotes/notes/deprecate-datadog-env-3011aafe83b66e37.yaml]: That linked worked, latest artifact shows the link to the public docs on Unified Service Tagging: https://845704-61572326-gh.circle-artifacts.com/0/tmp/docs/release_notes.html ! 👍

**Review by brettlangdon — APPROVED** [review]

File: ddtrace/bootstrap/sitecustomize.py (hunk 3 above)
Comment (brettlangdon) [reply ddtrace/bootstrap/sitecustomize.py:113]: agreed that this isn't super necessary here. `DD_ENV=""` would be weird anyways. We can skip making any changes to this for now.

**Pull request merged**